main/src/ca/uwaterloo/flix/MutationTester.scala

```diff
@@ -0,0 +1,278 @@
+                val cRes = flix.codeGen(newRoot).unsafeGet
+                val testResults = cRes.getTests.values.forall(c =>
+                    try {
+                        c.run() match {
+                            case java.lang.Boolean.TRUE => true
+                            case _ => false
+                        }
+                    } catch {
+                        case _: Throwable => false
+                    }
+                )
+                timeTemp += (System.nanoTime() - start).toFloat / 1000000000
+                if (testResults) {
+                    survivorCount += 1
+                    println("mutation in " + mDef.sym.toString + " survived")
+                }
+            })
+        })
+        val totalEndTime = System.nanoTime() - totalStartTime
+        println("there where " + survivorCount.toString + " surviving mutations, out of " + mutantCounter.toString + " mutations")
+        println("average time to test a mutant: " + timeTemp/mutantCounter)
+        println("total time to test all mutants: " + totalEndTime.toFloat / 1000000000)
+    }
+    def mutate(root: TypedAst.Root, testee: String): List[(Symbol.DefnSym, List[TypedAst.Def])] = {
+        val defs = root.defs
+        val defSyms = root.modules.filter(pair => (pair._1.toString.equals(testee))).values.toList.flatten
+        println(defSyms)
+        val mutatedDefs: List[Option[(Symbol.DefnSym, List[TypedAst.Def])]] = defs.toList.map(d => (d._1,d._2) match {
+            case (s, fun) =>
+                if (defSyms.contains(s)) {
+                    val mutExps = mutateExpr(fun.exp)
+                    val mutDefs = mutExps.map(mexp => TypedAst.Def(fun.sym, fun.spec, mexp))
+                    Some(d._1 -> mutDefs)
+                } else None
+            case _ => None
+        })
+        mutatedDefs.flatten
+    }
+
+
+    def mutateExpr(e: TypedAst.Expr): List[TypedAst.Expr] = {
+        e match {
+            case Expr.Cst(cst, tpe, loc) => Expr.Cst(mutateCst(cst), tpe, loc) :: Nil
+            case Expr.Binary(sop, exp1, exp2, typ, eff, loc) =>
+                val mut1 = Expr.Binary(mutateSop(sop), exp1, exp2, typ, eff, loc)
+                val mut2 = mutateExpr(exp1).map(m => Expr.Binary(sop, m, exp2, typ, eff, loc))
+                val mut3 = mutateExpr(exp2).map(m => Expr.Binary(sop, exp1, m, typ, eff, loc))
+                mut1::mut2:::mut3
+            case Expr.IfThenElse(exp1, exp2, exp3,typ, eff, loc) =>
+                val mut1 = mutateExpr(exp1).map(m => Expr.IfThenElse(m, exp2, exp3, typ, eff, loc))
+                val mut2 = mutateExpr(exp2).map(m => Expr.IfThenElse(exp1, m, exp3, typ, eff, loc))
+                val mut3 = mutateExpr(exp3).map(m => Expr.IfThenElse(exp1, exp2, m, typ, eff, loc))
+                mut1:::mut2:::mut3
+            // var doesn't contain a subtree, and we don't mutate them so we don't need to return them
+            case Expr.Var(_, _, _) => Nil
+            case e => List(e)
```

Comment:
I would try to get all expression filled in. Even if they just do some recursive calls or return themselves. This will give you a better overview.